src/cloud/benchflow/sender/sender.go

```diff
@@ -153,51 +180,74 @@ func consumeFromTopic(t TransformerSetting) {
 						panic(err)
 						}
 					fmt.Println("Script "+s.Script+" processed")
-					launchAnalyserScript(msg.Trial_id, msg.Total_trials_num, t.Topic)
+					totalTrialsNum, _ := strconv.Atoi(msg.Total_trials_num)
+					launchAnalyserScripts(msg.Trial_id, totalTrialsNum, t.Topic)
 					}
+				consumer.CommitUpto(m)
 			}
-			waitGroup.Done()
+		consumer.Close()
+		waitGroup.Done()
 		}()
 }
 
-func launchAnalyserScript(trialID string, totalTrials int, req string) {
+func launchAnalyserScripts(trialID string, totalTrials int, req string) {
 	var scripts []AnalyserScript
 	for _, s := range c.AnalysersSettings {
 		if s.Requirements == req {
 				scripts = s.Scripts
+				break
 			}
 		}
 	for _, s := range scripts {
-		var args []string
-		args = append(args, "--master", "local[*]")
-		args = append(args, "--packages", "TargetHolding:pyspark-cassandra:0.2.2")
-		args = append(args, s.TrialScript)
-		args = append(args, "local[*]")
-		args = append(args, os.Getenv("CASSANDRA_IP"))
-		args = append(args, trialID)
-		fmt.Println(args)
-		cmd := exec.Command(sparkHome+"/bin/spark-submit", args...)
-		cmd.Stdout = os.Stdout
-    	cmd.Stderr = os.Stderr
-		err := cmd.Start()
-		cmd.Wait()
-		if err != nil {
-			panic(err)
-			}
-		fmt.Println("Script "+s.TrialScript+" processed")
-		expID := strings.Split(trialID, "_")[0]
-		mutex.Lock()
-		_, present := trialCount[expID]
-		if(present) {
-			trialCount[expID] += 1
-			} else {
-			trialCount[expID] = 1
-			}
-		if(trialCount[expID] == totalTrials) {
-			// Launch Experiment metric
-			fmt.Printf("All trials for experiment "+expID+" completed")
-			}
-		mutex.Unlock()
+		go func(sc AnalyserScript) {
+			var args []string
+			args = append(args, "--master", "local[*]")
+			args = append(args, "--packages", "TargetHolding:pyspark-cassandra:0.2.2")
+			args = append(args, sc.TrialScript)
+			args = append(args, "local[*]")
+			args = append(args, os.Getenv("CASSANDRA_IP"))
+			args = append(args, trialID)
+			fmt.Println(args)
+			cmd := exec.Command(sparkHome+"/bin/spark-submit", args...)
+			cmd.Stdout = os.Stdout
+	    	cmd.Stderr = os.Stderr
+			err := cmd.Start()
+			cmd.Wait()
+			if err != nil {
+				panic(err)
+				}
+			fmt.Println("Script "+sc.TrialScript+" processed")
+			expID := strings.Split(trialID, "_")[0]
+			mutex.Lock()
+			_, present := trialCount[expID+":"+sc.TrialScript]
+			if(present) {
+				trialCount[expID+":"+sc.TrialScript] += 1
+				} else {
+				trialCount[expID+":"+sc.TrialScript] = 1
+				}
+			if(trialCount[expID+":"+sc.TrialScript] == totalTrials) {
+				// Launch Experiment metric
+				fmt.Printf("All trials "+sc.TrialScript+" for experiment "+expID+" completed, launching experiment analyser")
+				var argss []string
```

Comment:
@VincenzoFerme I plan on improving the command builder pattern in general later, also to streamline passing arguments to the scripts.